### PR TITLE
Convert Asset Icons to Chakra UI

### LINKED ISSUE #9048: Migrate SVG icon files to Chakra
To be inline with the UI migration #6374, there is a collection of SVG assets that serve as icons that can also be converted to Chakra!

Aside from the approach to use the `Icon` component, which is good for use with icons from 3rd party packages, there is the [createIcon()](https://chakra-ui.com/docs/components/icon/usage#using-the-createicon-function) function which allows you to build the svg parts and supercharge the icon itself with Chakra props. The icon can then be imported into another part of the project as a Chakra component to be styled directly.

This can cut down on, say, the need to use descendant selecting in CSS to be able to alter any styling.

> Note: using `createIcon()` over the `Icon` component is a semantic choice. From the Chakra docs, it states that `"[This] function is a convenience wrapper around the process of generating icons with 'Icon', allowing you to achieve the same functionality with less effort."`
>
> I personally find `createIcon()` to be cleaner for this use! 😅 

## Current Behavior
For example, the current approach is to import an svg as a react component via `gatsby-plugin-react-svg` as the following:
```ts
import HardwareSupport from "../../../assets/wallets/hardware_support.svg"
```

Now if the `fill` or any other styling of the `svg` element were need to be altered, you would use a descendant selector in the parent component
```tsx
// Using Chakra's `Box` component

<Box
  sx={{
    svg: {
      boxSize: 7,
      mt: 2,
      path: {
        fill: "text",
      },
    },
  }}
>
  <HardwareSupport />
</Box>
```

## New Behavior
With `createIcon()` for [hardware_support.svg](https://github.com/ethereum/ethereum-org-website/blob/dev/src/assets/wallets/hardware_support.svg?short_path=a2d4ef3):

```tsx
// Note the use of `fill="currentColor"` compared to the original

export const HardwareSupportIcon = createIcon({
  displayName: "HardwarSupportIcon",
  viewBox: "0 0 256 256",
  path: [
    <path
      d="M99.0003 42C100.415 42 101.772 42.5621 102.773 43.5626L130.913 71.7034C132.997 73.7869 132.997 77.1648 130.913 79.2483C128.83 81.3317 125.452 81.3317 123.368 79.2482L104.335 60.2149V107.786C104.335 110.732 101.947 113.121 99.0003 113.121C96.0538 113.121 93.6653 110.732 93.6653 107.786V60.2148L74.6318 79.2483C72.5484 81.3317 69.1704 81.3317 67.087 79.2483C65.0036 77.1648 65.0036 73.7869 67.087 71.7034L95.2278 43.5626C96.2283 42.5621 97.5853 42 99.0003 42Z"
      fill="currentColor"
    />,
    <path
      d="M169 48.0001C171.947 48.0001 174.335 50.3887 174.335 53.3351V100.906L193.368 81.8726C195.452 79.7891 198.83 79.7891 200.913 81.8726C202.997 83.956 202.997 87.3339 200.913 89.4174L172.773 117.558C171.772 118.559 170.415 119.121 169 119.121C167.585 119.121 166.228 118.559 165.228 117.558L137.087 89.4174C135.004 87.334 135.004 83.956 137.087 81.8726C139.17 79.7891 142.548 79.7891 144.632 81.8726L163.665 100.906V53.3351C163.665 50.3887 166.054 48.0001 169 48.0001Z"
      fill="currentColor"
    />,
    <path
      fill-rule="evenodd"
      clip-rule="evenodd"
      d="M70.6984 205.951C68.1036 205.951 66 203.857 66 201.274V178.387C66 175.804 68.1035 173.71 70.6984 173.71H86.0249V167.067H126.725L109.176 136.809C107.878 134.572 108.648 131.711 110.895 130.42L143.01 111.963C145.258 110.671 148.131 111.437 149.429 113.675L186.883 178.251C193.3 189.316 189.492 203.465 178.376 209.853C176.874 210.717 175.316 211.394 173.728 211.892C171.207 212.825 168.479 213.335 165.631 213.335H86.0249V205.951H70.6984ZM73.9135 181.588H86.0249V198.074H73.9135V181.588ZM165.631 167.067C167.739 167.067 169.782 167.346 171.724 167.87L144.183 120.385L117.637 135.642L135.863 167.067H165.631Z"
      fill="currentColor"
    />,
  ],
})
```

The original example can now become:
```ts
import { HardwareSupportIcon } from "../../../components/icons/wallets/HardwareSupportIcon"

// Or exported through a common `index.ts` for each sub-directory
import { HardwareSupportIcon } from "../../../components/icons/wallets"
```

```tsx
<Box>
  <HardwareSupportIcon color="text" boxSize={ 7 } mt={ 2 }  />
</Box>
```

## Setup

As this is arguably a major overhaul, the following are proposed outlined steps to be taken:
1. Identify all svg icons from the `assets` directory
2. Convert each icon to a `.tsx` file with Chakra's `createIcon()` function, save the new files into `src/components/icons`, and divided here in the same sub-directory structure present in `assets`. Also, they would be given a new naming convention of `[Name]Icon.tsx`
  - i.e. `hardware_support.svg` currently resides in `src/assets/wallets/hardware_support.svg`. Its name and location would now become `src/components/icons/wallets/HardwareSupportIcon.tsx`
3. Convert the icon imports in all components and pages.
3a. Adjust implementation as necessary.

## Additional Thoughts
One known expected result in this conversion is the loss of SVG image previews via an IDE extension, and a solid replacement was not found. However, with the current consideration of bringing in StorybookJS to the workflow a story could be created to display all of the icons.

### PR DESCRIPTION
**Pull request by TylerAPfledderer**
## Description

This migrates all `*.svg` icons to Chakra and `*.tsx` files.

###  Notes:

The following are migrated but have been identified as not currently in use.
- `assets/highlight.svg`
- `assets/highlight-dark.svg`
- `assets/run-a-node/decentralized-eth-glyph` (appears to be a duplicate of `decentralized-glyph.svg`)
- `assets/wallets/frame.svg`

Should `fontSize` be utilized instead of width and height or `boxSize`?

❗ Should the naming of these new files not include `Glyph`?

## Related Issue
Closes #9048

### CODE REVIEW

**Review by hoanglamcoin — APPROVED** [review]

**Comment by nloureiro** [comment]
@TylerAPfledderer can you help me understand a little more this?

Do you want to port all the SVG icons to the chakra icon component?
https://chakra-ui.com/docs/components/icon

I'm asking because, in the Design system plan, we aim to build a new ethereum.org icon family somewhere in the future. Something open source and web3 packed.

Would this issue be the beginning?

**Comment by peetzweg** [comment]
Uff that was probably a lot of work 👍 

IMHO I think it a bit weird copying the svg code into the tsx files. Isn't the bundler able to import svg files directly and just pass this as the `path` prop for `createIcon()`?

Feels hard to maintain in the future if the actual SVG is hardcoded in a tsx file. No vector tool can open `.tsx` files to edit it.

Furthermore the svg code could maybe use some optimization with something like [scour](https://github.com/scour-project/scour) to reduce precision to save on file size.

**Comment by TylerAPfledderer** [comment]
@nloureiro 
> Do you want to port all the SVG icons to the chakra icon component?
https://chakra-ui.com/docs/components/icon

That is the intention here. 

This proposed change did not take into account a new set of SVGs with the intended purpose you describe. I was not aware of that goal, and would like to know more about what you mean in the new collection being web3 packed. 😄 

---

@peetzweg
> IMHO I think it a bit weird copying the svg code into the tsx files. Isn't the bundler able to import svg files directly and just pass this as the path prop for createIcon()?

The `path` prop is meant for the internal elements of the svg, as by default it creates the svg element. So importing the SVG to the function that way would render a nested svg

The `createIcon()` function makes the svg directly supercharged with Chakra, so you can supply Chakra style props and theming on it. This also eliminates the need to use a parent element in some cases to style it via descendant selecting, say.

Indeed, you can instead pass the svg file to Chakra's `Icon` component via the `as` prop and be able to style that way. `createIcon` could give you a little more control in how you can alter the styling of the internals. 

I would use `Icon` to handle third-party icons, and `createIcon()` for internal icons.

As far as management, I come from a standpoint of not having designed SVGs. I would think that the process here would be to create the svgs, send them to an app like SVGOmg to optimize the code, transfer the code setup to `createIcon()`, test in the environment (like with Storybook since that is going to used going forward), and if there is any issue revisit the SVG file rinse and repeat as needed!

The expectation then is that the original svg assets are saved if changes need to be addressed in the design. Though I am open to understanding if this would not be an expected or desirable process; I'm all for getting a better idea of the workflow! 😃 🤘🏼

**Comment by nloureiro** [comment]
> 

@TylerAPfledderer this goal is more a collateral of the Design system.

Having an open-source design system allows us to give others the ability to work on top of our work.

The icons are the best example of a small part of the design system that can be used separably, and with that in mind, I have been building the idea of creating an ethereum.org icon family.

It would be based on the design style we use on ethereum.org and can be updated and transformed by the community for their own projects.

On top of the normal icons, we could have more web3 icons that most icon families don't have, like the ones I did for the find my wallet page.

Screen Shot 2023-01-16 09 56 59 AM

Does it help? In the way that you are thinking of converting the SVG to an icon?

**Comment by TylerAPfledderer** [comment]
@nloureiro thank you for elaborating!

In that case, the new library you looking to building should be treated like 3rd-party icons and use the `Icon` component instead with the `as` prop.
```tsx
<Icon as={SomeIcon} {...stylePropsHere} />
``` 
> We can still use the Chakra theme config to handle any base styles if need be!

This way -- barring other usage that would prevent you from doing this -- the SVGs might need to be optimized so there is no need to drill down to the internal elements via a descendant selector to alter colors, unless there is an edge case or if an SVG might have some small animations going on. (I've only seen one or two instances where this was occurring so maybe this a non-issue 😄 )

**Comment by nloureiro** [comment]
> @nloureiro thank you for elaborating!
> 
> In that case, the new library you looking to building should be treated like 3rd-party icons and use the `Icon` component instead with the `as` prop.
> 
> ```tsx
> <Icon as={SomeIcon} {...stylePropsHere} />
> ```
> 
>  > We can still use the Chakra theme config to handle any base styles if need be!
> 
> This way -- barring other usage that would prevent you from doing this -- the SVGs might need to be optimized so there is no need to drill down to the internal elements via a descendant selector to alter colors, unless there is an edge case or if an SVG might have some small animations going on. (I've only seen one or two instances where this was occurring so maybe this a non-issue 😄 )


Ok, got it. Makes sense to use it as a 3rd party 'icon'

Question: having this in mind, does it make sense to keep this issue open? Should we start the new icons collection with this svg's? or am I seeing wrong and we will add this later when implementing the Design system

**Comment by TylerAPfledderer** [comment]
I have no objection to closing this PR and the respective issue.

I'd like to know what @pettinarip thinks about next steps here. 😁

**Review by pettinarip — CHANGES_REQUESTED** [review]
Thanks a lot for the time and effort you have put on this @TylerAPfledderer 

I've only found a small issue on the staking page.

======

@nloureiro and I talked about this yesterday and we've decided to include these changes since they will be part of the transition over to the icons font in the near future.

Once we have the font, we'll start deprecating these icons and removing them. In the meanwhile, we can benefit from these newly wrapped icons :)

> Should the naming of these new files not include Glyph?

As we are going to remove these icons at some point, I wouldn't mind doing that at this moment. I think its fine for now.

File: src/components/icons/staking/StakingGlyphCloudIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+export const StakingGlyphCloudIcon = createIcon({
+  displayName: "StakingGlyphCloudIcon",
+  viewBox: "0 0 50 28",
+  defaultProps: {
+    width: "50px",
+    height: "28px",
+  },
+  d: "M41.9655 12.2876C41.6352 5.44298 36.2483 0 29.6514 0C24.479 0 20.0505 3.34612 18.2199 8.09041C17.2547 7.68167 16.2003 7.45668 15.0959 7.45668C11.6055 7.45668 8.61465 9.70388 7.36264 12.8963C7.27461 12.8929 7.18617 12.8912 7.09734 12.8912C3.17759 12.8912 0 16.2297 0 20.3478C0 24.3385 2.98385 27.597 6.73542 27.7949V27.8045H43.3048V27.7699C47.061 27.3979 50 24.0755 50 20.0319C50 15.7392 46.6878 12.2593 42.6019 12.2593C42.3875 12.2593 42.1753 12.2688 41.9655 12.2876Z",
+})
```
[the remark this answers is not in the document]
Comment (TylerAPfledderer) [reply src/components/icons/staking/StakingGlyphCloudIcon.tsx:11]: This is coming from the `StakingHeirarchy` component and it's child component `Glyph` where the `md` was setting`boxSize` to `initial` which overrides the original width/height set from the icons themselves. Had to explicitly set the value to `50px` here.

**Comment by TylerAPfledderer** [comment]
@pettinarip we should be good here! A couple of merge conflicts that were resolved. (Still had some trouble handling them properly!)

I'll leave alone the file names with `Glyph`.

**Review by pettinarip — APPROVED** [review]
Niceee 🚀

**Review by wackerow — APPROVED** [review]
Phew, that's a lot of SVGs, nice work @TylerAPfledderer!

I do think I would be nice to remove some of the redundant naming like "glyph" and "icon", and maybe placing those items into a folder labeled as such instead (ie /glyphs/Decentralization). But, rather than complicating this PR further I would suggest we address that separately if we want to do that. This looks good to go from my perspective.

Will just make note that for the /assets page, any SVG's we want to host we'll need to have a publicly available static `.svg` file for folks to download since these won't be available.

**Comment by pettinarip** [comment]
> Will just make note that for the /assets page, any SVG's we want to host we'll need to have a publicly available static .svg file for folks to download since these won't be available.

Agreed. Just to let you know, the svgs that lives under `src/assets/assets` are still in there, those were not deleted or modified. Those svgs are the ones being consumed in the `/assets` page. This is related to this PR #8891 where we are adding new svgs to be downloaded.

**Review by corwintines — APPROVED** [review]
Nice one @TylerAPfledderer. I just merged dev to fix some conflicts, but this is good to go 🎉

**Pull request merged**